tap_universal_file/streams.py

```diff
@@ -0,0 +1,463 @@
+            file_name = file["name"]
+            if self.config["delimited_delimiter"] == "detect":
+                if re.match(".*\\.csv.*", file_name):
+                    delimiter = ","
+                elif re.match(".*\\.tsv.*", file_name):
+                    delimiter = "\t"
+                else:
+                    msg = (
+                        "Configuration option 'delimited_delimiter' is set to 'detect' "
+                        "but a non-csv non-tsv file is present. Please manually "
+                        "specify 'delimited_delimiter'."
+                    )
+                    raise RuntimeError(msg)
+            else:
+                delimiter = self.config["delimited_delimiter"]
+
+            yield (
+                self.ModifiedDictReader(
+                    f=self._skip_rows(file_name),
+                    delimiter=delimiter,
+                    quotechar=quote_character,
+                    fieldnames=override_headers,
+                    config=self.config,
+                ),
+                file_name,
+                file["last_modified"],
+            )
+
+    def _skip_rows(self, file: str) -> list[str]:
+        """Takes a file name and only returns rows which are not configured as skipped.
+
+        Args:
+            file: The name of the file to process.
+
+        Returns:
+            A list containing rows from the file.
+        """
+        with self.fs_manager.filesystem.open(
+            path=file,
+            mode="rt",
+            compression=self.get_compression(file=file),
+        ) as f:
+            file_list = []
+            file_list.extend(f)
+        try:
+            for _ in range(self.config["delimited_header_skip"]):
+                file_list.pop(0)
+            for _ in range(self.config["delimited_footer_skip"]):
+                file_list.pop()
+        except IndexError:
+            return []
+        else:
+            return file_list
+
+    class ModifiedDictReader(csv.DictReader):
+        """A modified version of DictReader that detects improperly formatted rows."""
+
+        def __init__(  # noqa: PLR0913
+            self,
+            f: Any,  # noqa: ANN401
+            fieldnames: Any | None = None,
+            restkey: Any | None = None,
+            restval: Any | None = None,
+            dialect: str = "excel",
+            config: dict = None,
+            *args: Any,
+            **kwds: Any,
+        ) -> None:
+            """Identical to the superclass's method except for defining self.config."""
+            super().__init__(f, fieldnames, restkey, restval, dialect, *args, **kwds)
+            self.config = config if config is not None else {}
+
+        def __next__(self) -> dict:
+            """Identical to the superclass's method except for raising FileFormatErrors.
+
+            Raises:
+                FileFormatError: If a row in the *SV has too few entries.
```

Comment:
Updated the docs rather than updating the exception. My reasoning is that a RuntimeError is probably fine here as long as the error message gives appropriately specific information, which it does.